Add doco for customized outbox isolation level (#5229)
Particular/docs.particular.net: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md

In addition to that connection managed by NServiceBus, users can open their own database connections in the message handlers. If the underlying database technology supports distributed transactions managed by Microsoft Distributed Transaction Coordinator -- MS DTC (e.g. SQL Server, Oracle or PostgreSQL), the transaction gets escalated to a distributed transaction.

> The `TransactionScope` mode is most useful in legacy scenarios e.g. when migrating from MSMQ transport to a messaging infrastructure that does not support MS DTC. In order to maintain consistency the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.

Reviewer: ```suggestion
The `TransactionScope` mode is most useful in legacy scenarios such as when migrating from MSMQ transport to a messaging infrastructure that does not support MSDTC. In order to maintain consistency, the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.
```

> When necessary, the transaction isolation level can also be adjusted. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher isolation level to guarantee correctness (e.g. `RepeatableRead` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:

Reviewer 2: Does this change the isolation level of the whole outbox transaction? What is the default?

> When necessary, the transaction isolation level can also be adjusted. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher isolation level to guarantee correctness (e.g. `RepeatableRead` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:

Author: ```suggestion
When necessary, the outbox transaction isolation level can also be adjusted. The change affects all data access done within this transaction. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher than default (`Read Committed`) isolation level to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
```

> When necessary, the outbox transaction isolation level can also be adjusted. The change affects all data access done within this transaction. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher than default (`Read Committed`) isolation level to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:

Reviewer: ```suggestion
When necessary, the outbox transaction isolation level can also be adjusted. The change in the isolation level affects all data access operations done within the transaction. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher than default (`Read Committed`) isolation level to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
```

snippet: SqlPersistenceOutboxIsolationLevel

> Note: Adjusting the isolation level requires version 6.1 of SQL Persistence.

Reviewer 2: Could this section be a separate `[6.1,)` snippet?

```suggestion
Note: Adjusting the isolation level requires version 6.1 of SQL Persistence or higher.
```
